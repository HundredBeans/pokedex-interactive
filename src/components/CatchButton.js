import React, { useState } from "react";
import {
  Box,
  Image,
  Button,
  Popover,
  PopoverContent,
  PopoverHeader,
  PopoverCloseButton,
  PopoverBody,
  PopoverFooter,
  ButtonGroup,
  PopoverTrigger,
  Progress,
  FormControl,
  FormLabel,
  Input,
  // useBreakpointValue,
} from "@chakra-ui/react";
// import { Formik } from "formik";
import pokeball from "../assets/pokeball.png";

function CatchButton(props) {
  const [isOpen, setIsOpen] = useState(false);
  const open = () => {
    setIsOpen(!isOpen);
  };
  const closePopover = () => {
    setIsOpen(false);
  };
  return (
    <React.Fragment>
      <Box
        bottom="0"
        left="50%"
        transform="translate(-50%, 0);"
        position="fixed"
        zIndex="banner"
        shadow="lg"
        borderRadius="full"
      >
        <Button
          variant="ghost"
          boxSize="100px"
          width="100%"
          rounded="full"
          onClick={open}
        >
          <Image src={pokeball} />
        </Button>
      </Box>
      <Popover isOpen={isOpen} onClose={closePopover} closeOnBlur={false}>
        <PopoverTrigger>
          <Box bottom="100px" right="50%" position="fixed"></Box>
        </PopoverTrigger>
        <PopoverContent width="100%" shadow="dark-lg">
          {props.catchSuccess ? (
            <CatchSuccessContent {...props} />
          ) : (
            <InitialContent close={closePopover} {...props} />
          )}
        </PopoverContent>
      </Popover>
    </React.Fragment>
  );
}

const InitialContent = (props) => {
  return (
    <React.Fragment>
      <PopoverHeader fontWeight="semibold" textTransform="capitalize">
        Catch {props.pokemonName}
      </PopoverHeader>
      <PopoverCloseButton />
      <PopoverBody>
        {props.catchLoading ? (
          <Progress size="xs" isIndeterminate />
        ) : (
          <React.Fragment>
            Are you sure you want to catch this Pokemon?{" "}
            <strong>(Success probability {props.catchProbability}%)</strong>
          </React.Fragment>
        )}
      </PopoverBody>
      <PopoverFooter d="flex" justifyContent="flex-end">
        <ButtonGroup size="sm">
          <Button
            variant="outline"
            onClick={props.close}
            disabled={props.catchLoading}
          >
            Cancel
          </Button>
          <Button
            colorScheme="teal"
            onClick={props.catchPokemon}
            isLoading={props.catchLoading}
            loadingText="Catching..."
          >
            Catch
          </Button>
        </ButtonGroup>
      </PopoverFooter>
    </React.Fragment>
  );
};

const SetNicknameForm = (props) => {
  const [nickname, setNickname] = useState("");
  const handleChange = (e) => {
    setNickname(e.target.value);
  };
  return (
    <React.Fragment>
      <PopoverBody>
        <FormControl>
          <FormLabel htmlFor="nickname" textTransform="capitalize">
            Your {props.pokemonName}&apos;s Nickname
          </FormLabel>
          <Input id="nickname" onChange={handleChange} />
        </FormControl>
      </PopoverBody>
      <PopoverFooter d="flex" justifyContent="flex-end">
        <ButtonGroup size="sm">
          <Button
            colorScheme="teal"
            isLoading={props.isSaving}
            loadingText="Saving..."
            disabled={!nickname}
            onClick={() => props.saveNickname(nickname)}
          >
            Save
          </Button>
        </ButtonGroup>
      </PopoverFooter>
    </React.Fragment>
  );
};

const CatchSuccessContent = (props) => {
  return (
    <React.Fragment>
      <PopoverHeader fontWeight="semibold" textTransform="capitalize">
        Gotcha!
      </PopoverHeader>
      <PopoverCloseButton />
      <SetNicknameForm {...props} />
    </React.Fragment>
  );
};

export default CatchButton;
