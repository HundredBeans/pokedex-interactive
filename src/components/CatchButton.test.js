import { cleanup, fireEvent, waitFor } from "@testing-library/react";
import React from "react";
import { render } from "../test-utils";
import CatchButton from "./CatchButton";

describe("CatchButton.js test", () => {
  afterEach(() => {
    cleanup();
  });

  it("renders the component (Initial Content)", () => {
    const renderedComponent = render(<CatchButton catchSuccess={false} />);
    expect(renderedComponent).toMatchSnapshot("initial-content");
  });

  it("renders the component (CatchSuccess Content)", () => {
    const renderedComponent = render(<CatchButton catchSuccess={true} />);
    expect(renderedComponent).toMatchSnapshot("catchSuccess-content");
  });

  it("renders the component (CatchLoading Content)", () => {
    const renderedComponent = render(
      <CatchButton catchSuccess={false} catchLoading={true} />
    );
    expect(renderedComponent).toMatchSnapshot("catchLoading-content");
  });

  test("setNicknameForm working correctly", async () => {
    const saveNickname = jest.fn();
    const pokemonName = "bulbasaur";
    const nickname = "Bulbasaur Nickname";
    const renderedComponent = render(
      <CatchButton
        catchSuccess={true}
        pokemonName={pokemonName}
        isSaving={false}
        saveNickname={saveNickname}
      />
    );
    const nicknameFormInput = renderedComponent.getByLabelText(
      `Your ${pokemonName}'s Nickname`
    );
    const saveButton = renderedComponent.getByText("Save");
    // Check if the button is disabled at first
    expect(saveButton).toBeDisabled();
    fireEvent.change(nicknameFormInput, { target: { value: nickname } });
    await waitFor(() => {
      // Check if the button is not disabled
      expect(saveButton).not.toBeDisabled();
      expect(nicknameFormInput.value).toBe(nickname);
      fireEvent.click(saveButton);
      expect(saveNickname).toHaveBeenCalled();
    });
  });
});
