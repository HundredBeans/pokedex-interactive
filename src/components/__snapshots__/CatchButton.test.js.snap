// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatchButton.js test renders the component (CatchLoading Content): catchLoading-content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="css-1ue797s"
      >
        <button
          class="chakra-button css-17i1qg7"
          type="button"
        >
          <img
            class="chakra-image__placeholder css-0"
          />
        </button>
      </div>
      <div
        aria-controls="popover-content-15"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="css-1rpneay"
        id="popover-trigger-15"
      />
      <div
        class="css-11r8ny"
        data-popper-escaped=""
        data-popper-placement="bottom"
        data-popper-reference-hidden=""
        style="--popper-arrow-shadow-color: #E2E8F0; visibility: hidden; margin: 0px; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; transform: translate(0px, 8px);"
      >
        <section
          aria-describedby="popover-body-15"
          aria-labelledby="popover-header-15"
          class="chakra-popover__content css-l1sj48"
          id="popover-content-15"
          role="dialog"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
          tabindex="-1"
        >
          <header
            class="chakra-popover__header css-1x1js2i"
            id="popover-header-15"
          >
            Catch 
          </header>
          <button
            aria-label="Close"
            class="css-y3dsx3"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="chakra-popover__body css-1ews2c8"
            id="popover-body-15"
          >
            <div
              class="css-n21wkn"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                class="css-p242e5"
                data-indeterminate=""
                role="progressbar"
                style="width: 0%;"
              />
            </div>
          </div>
          <footer
            class="chakra-popover__footer css-1brmfnp"
          >
            <div
              class="chakra-button__group css-1ezzth2"
              role="group"
            >
              <button
                class="chakra-button css-1u73en3"
                disabled=""
                type="button"
              >
                Cancel
              </button>
              <button
                class="chakra-button css-1k6fjms"
                data-loading=""
                disabled=""
                type="button"
              >
                <div
                  class="chakra-button__spinner css-149izd1"
                >
                  <div
                    class="chakra-spinner css-1ojda53"
                  >
                    <span
                      class="css-f8n5zr"
                    >
                      Loading...
                    </span>
                  </div>
                </div>
                Catching...
              </button>
            </div>
          </footer>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1ue797s"
    >
      <button
        class="chakra-button css-17i1qg7"
        type="button"
      >
        <img
          class="chakra-image__placeholder css-0"
        />
      </button>
    </div>
    <div
      aria-controls="popover-content-15"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="css-1rpneay"
      id="popover-trigger-15"
    />
    <div
      class="css-11r8ny"
      data-popper-escaped=""
      data-popper-placement="bottom"
      data-popper-reference-hidden=""
      style="--popper-arrow-shadow-color: #E2E8F0; visibility: hidden; margin: 0px; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; transform: translate(0px, 8px);"
    >
      <section
        aria-describedby="popover-body-15"
        aria-labelledby="popover-header-15"
        class="chakra-popover__content css-l1sj48"
        id="popover-content-15"
        role="dialog"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
        tabindex="-1"
      >
        <header
          class="chakra-popover__header css-1x1js2i"
          id="popover-header-15"
        >
          Catch 
        </header>
        <button
          aria-label="Close"
          class="css-y3dsx3"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="chakra-popover__body css-1ews2c8"
          id="popover-body-15"
        >
          <div
            class="css-n21wkn"
          >
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              class="css-p242e5"
              data-indeterminate=""
              role="progressbar"
              style="width: 0%;"
            />
          </div>
        </div>
        <footer
          class="chakra-popover__footer css-1brmfnp"
        >
          <div
            class="chakra-button__group css-1ezzth2"
            role="group"
          >
            <button
              class="chakra-button css-1u73en3"
              disabled=""
              type="button"
            >
              Cancel
            </button>
            <button
              class="chakra-button css-1k6fjms"
              data-loading=""
              disabled=""
              type="button"
            >
              <div
                class="chakra-button__spinner css-149izd1"
              >
                <div
                  class="chakra-spinner css-1ojda53"
                >
                  <span
                    class="css-f8n5zr"
                  >
                    Loading...
                  </span>
                </div>
              </div>
              Catching...
            </button>
          </div>
        </footer>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CatchButton.js test renders the component (CatchSuccess Content): catchSuccess-content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="css-1ue797s"
      >
        <button
          class="chakra-button css-17i1qg7"
          type="button"
        >
          <img
            class="chakra-image__placeholder css-0"
          />
        </button>
      </div>
      <div
        aria-controls="popover-content-8"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="css-1rpneay"
        id="popover-trigger-8"
      />
      <div
        class="css-11r8ny"
        data-popper-escaped=""
        data-popper-placement="bottom"
        data-popper-reference-hidden=""
        style="--popper-arrow-shadow-color: #E2E8F0; visibility: hidden; margin: 0px; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; transform: translate(0px, 8px);"
      >
        <section
          aria-describedby="popover-body-8"
          aria-labelledby="popover-header-8"
          class="chakra-popover__content css-l1sj48"
          id="popover-content-8"
          role="dialog"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
          tabindex="-1"
        >
          <header
            class="chakra-popover__header css-1x1js2i"
            id="popover-header-8"
          >
            Gotcha!
          </header>
          <button
            aria-label="Close"
            class="css-y3dsx3"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="chakra-popover__body css-1ews2c8"
            id="popover-body-8"
          >
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <label
                class="chakra-form__label css-1wjdhu5"
                for="nickname"
                id="field-9-label"
              >
                Your 
                's Nickname
              </label>
              <input
                class="chakra-input css-1b73kmm"
                id="nickname"
              />
            </div>
          </div>
          <footer
            class="chakra-popover__footer css-1brmfnp"
          >
            <div
              class="chakra-button__group css-1ezzth2"
              role="group"
            >
              <button
                class="chakra-button css-1k6fjms"
                disabled=""
                type="button"
              >
                Save
              </button>
            </div>
          </footer>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1ue797s"
    >
      <button
        class="chakra-button css-17i1qg7"
        type="button"
      >
        <img
          class="chakra-image__placeholder css-0"
        />
      </button>
    </div>
    <div
      aria-controls="popover-content-8"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="css-1rpneay"
      id="popover-trigger-8"
    />
    <div
      class="css-11r8ny"
      data-popper-escaped=""
      data-popper-placement="bottom"
      data-popper-reference-hidden=""
      style="--popper-arrow-shadow-color: #E2E8F0; visibility: hidden; margin: 0px; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; transform: translate(0px, 8px);"
    >
      <section
        aria-describedby="popover-body-8"
        aria-labelledby="popover-header-8"
        class="chakra-popover__content css-l1sj48"
        id="popover-content-8"
        role="dialog"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
        tabindex="-1"
      >
        <header
          class="chakra-popover__header css-1x1js2i"
          id="popover-header-8"
        >
          Gotcha!
        </header>
        <button
          aria-label="Close"
          class="css-y3dsx3"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="chakra-popover__body css-1ews2c8"
          id="popover-body-8"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1wjdhu5"
              for="nickname"
              id="field-9-label"
            >
              Your 
              's Nickname
            </label>
            <input
              class="chakra-input css-1b73kmm"
              id="nickname"
            />
          </div>
        </div>
        <footer
          class="chakra-popover__footer css-1brmfnp"
        >
          <div
            class="chakra-button__group css-1ezzth2"
            role="group"
          >
            <button
              class="chakra-button css-1k6fjms"
              disabled=""
              type="button"
            >
              Save
            </button>
          </div>
        </footer>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CatchButton.js test renders the component (Initial Content): initial-content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="css-1ue797s"
      >
        <button
          class="chakra-button css-17i1qg7"
          type="button"
        >
          <img
            class="chakra-image__placeholder css-0"
          />
        </button>
      </div>
      <div
        aria-controls="popover-content-2"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="css-1rpneay"
        id="popover-trigger-2"
      />
      <div
        class="css-11r8ny"
        data-popper-escaped=""
        data-popper-placement="bottom"
        data-popper-reference-hidden=""
        style="--popper-arrow-shadow-color: #E2E8F0; visibility: hidden; margin: 0px; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; transform: translate(0px, 8px);"
      >
        <section
          aria-describedby="popover-body-2"
          aria-labelledby="popover-header-2"
          class="chakra-popover__content css-l1sj48"
          id="popover-content-2"
          role="dialog"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
          tabindex="-1"
        >
          <header
            class="chakra-popover__header css-1x1js2i"
            id="popover-header-2"
          >
            Catch 
          </header>
          <button
            aria-label="Close"
            class="css-y3dsx3"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="chakra-popover__body css-1ews2c8"
            id="popover-body-2"
          >
            Are you sure you want to catch this Pokemon?
             
            <strong>
              (Success probability 
              %)
            </strong>
          </div>
          <footer
            class="chakra-popover__footer css-1brmfnp"
          >
            <div
              class="chakra-button__group css-1ezzth2"
              role="group"
            >
              <button
                class="chakra-button css-1u73en3"
                type="button"
              >
                Cancel
              </button>
              <button
                class="chakra-button css-1k6fjms"
                type="button"
              >
                Catch
              </button>
            </div>
          </footer>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1ue797s"
    >
      <button
        class="chakra-button css-17i1qg7"
        type="button"
      >
        <img
          class="chakra-image__placeholder css-0"
        />
      </button>
    </div>
    <div
      aria-controls="popover-content-2"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="css-1rpneay"
      id="popover-trigger-2"
    />
    <div
      class="css-11r8ny"
      data-popper-escaped=""
      data-popper-placement="bottom"
      data-popper-reference-hidden=""
      style="--popper-arrow-shadow-color: #E2E8F0; visibility: hidden; margin: 0px; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; transform: translate(0px, 8px);"
    >
      <section
        aria-describedby="popover-body-2"
        aria-labelledby="popover-header-2"
        class="chakra-popover__content css-l1sj48"
        id="popover-content-2"
        role="dialog"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
        tabindex="-1"
      >
        <header
          class="chakra-popover__header css-1x1js2i"
          id="popover-header-2"
        >
          Catch 
        </header>
        <button
          aria-label="Close"
          class="css-y3dsx3"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="chakra-popover__body css-1ews2c8"
          id="popover-body-2"
        >
          Are you sure you want to catch this Pokemon?
           
          <strong>
            (Success probability 
            %)
          </strong>
        </div>
        <footer
          class="chakra-popover__footer css-1brmfnp"
        >
          <div
            class="chakra-button__group css-1ezzth2"
            role="group"
          >
            <button
              class="chakra-button css-1u73en3"
              type="button"
            >
              Cancel
            </button>
            <button
              class="chakra-button css-1k6fjms"
              type="button"
            >
              Catch
            </button>
          </div>
        </footer>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
